### ISSUE TITLE
Rect::for_each is inclusive, vs. other functions exclusive

### ISSUE THREAD

**Issue by cg123** [amethyst/bracket-lib#230]
`Rect::for_each` iterates inclusively to (x2, y2), which disagrees with the convention used by all other Rect functions. It looks like the exclusive convention was decided on in 223148ea11ada7af45204cd35b8d0f6f8463dee1, but the `for_each` function was left as it was.

**Comment by dfannius** [comment]
I just came here to report the same thing. Note that `test_rect_set()` and `test_rect_callback()` actually explicitly test that `Rect::with_size(0, 0, 1, 1)` has one point when accessed with `Rect::point_set` but four points when accessed with `Rect::for_each`; I assume the test is wrong.

**Comment by thebracket** [comment]
I agree that this is out-of-character for Rust in general (I found inclusive vs. exclusive ranges confusing when I started Rust; it makes sense, my head kept switching the default for some reason). My concern is the knock-on effect of changing it - there's a few projects out there that use the old behavior, and I don't want to break them. I'll report back on this issue in a few when I've done some testing.
